liquidctl/driver/asetek_pro.py

```diff
@@ -0,0 +1,271 @@
+"""liquidctl drivers for sixth generation Asetek 690LC liquid coolers.
+
+Supported devices:
+
+- Corsair H100i PRO RGB, H115i PRO RGB or H150i PRO RGB (Experimental)
+
+Copyright (C) 2018–2020  Jonas Malaco and contributors
+
+Incorporates or uses as reference work by Kristóf Jakab, Sean Nelson
+and Chris Griffith.
+
+SPDX-License-Identifier: GPL-3.0-or-later
+"""
+
+import logging
+
+import usb
+
+from liquidctl.driver.asetek import Modern690Lc
+from liquidctl.driver.usb import UsbDriver
+from liquidctl.error import NotSupportedByDevice
+from liquidctl.keyval import RuntimeStorage
+from liquidctl.util import clamp
+
+
+LOGGER = logging.getLogger(__name__)
+
+_MAX_PROFILE_POINTS = 7
+_CRITICAL_TEMPERATURE = 60
+_HIGH_TEMPERATURE = 45
+_READ_LENGTH = 32
+_READ_TIMEOUT = 2000
+_WRITE_TIMEOUT = 2000
+
+_PRO_WRITE_ENDPOINT = 0x1
+_PRO_READ_ENDPOINT = 0x81
+_PRO_CMD_READ_FIRMWARE = 0xaa
+_PRO_CMD_READ_AIO_TEMP = 0xa9
+_PRO_CMD_READ_FAN_SPEED = 0x41
+_PRO_CMD_WRITE_FAN_SPEED = 0x40
+_PRO_CMD_READ_PUMP_SPEED = 0x31
+_PRO_CMD_READ_PUMP_MODE = 0x33
+_PRO_CMD_WRITE_PUMP_MODE = 0x32
+_PRO_PUMP_MODES = [
+    'Quiet',
+    'Balanced',
+    'Performance'
+    ]
+
+_PRO_SPEEDS = {
+    'rainbow': [
+        0x30, # slow
+        0x18, # medium
+        0x0C  # fast
+    ],
+    'shift': [
+        0x46, # slow
+        0x28, # medium
+        0x0F  # fast
+    ],
+    'pulse':[
+        0x50, # slow
+        0x37, # medium
+        0x1E  # fast
+    ],
+    'blinking': [
+        0x0F, # slow
+        0x0A, # medium
+        0x05  # fast
+    ]
+}
+
+
+class CorsairAsetekProDriver(Modern690Lc):
+    """liquidctl driver for Corsair-branded sixth generation Asetek coolers."""
+    SUPPORTED_DEVICES = [
+        (0x1b1c, 0x0c12, None, 'Corsair Hydro H150i Pro', {}),
+        (0x1b1c, 0x0c13, None, 'Corsair Hydro H115i Pro', {}),
+        (0x1b1c, 0x0c14, None, 'Corsair Hydro H110i Pro', {}),
+        (0x1b1c, 0x0c15, None, 'Corsair Hydro H100i Pro', {}),
+        (0x1b1c, 0x0c16, None, 'Corsair Hydro H80i Pro', {}),
+    ]
+
+    def _write(self, data):
+        """Write data to the AIO and log"""
+        LOGGER.debug('write %s', ' '.join(format(i, '02x') for i in data))
```

Comment:
See lazy hex strings in the style guide.

Reply:
Done